webpack/assets/javascripts/react_app/components/common/ClipboardCopy/index.js

```diff
@@ -11,16 +11,33 @@ const ClipboardCopy = ({
   buttonText,
   textareaProps,
   buttonProps,
+  hideTextarea,
+  withCopyIcon,
 }) => {
   const [text, setText] = useState(defaultText);
 
+  let clipboardClass = 'clipboard-copy';
+  let iconMarginLeft = '5px';
+
+  if (hideTextarea === true) {
+    clipboardClass = 'clipboard-copy-simple';
+  }
+
+  if (buttonText === '') {
+    iconMarginLeft = '0px';
+  }
+
   return (
-    <div className="clipboard-copy">
-      <textarea
-        defaultValue={text}
-        onChange={({ target: { value } }) => setText(value)}
-        {...textareaProps}
-      />
+    <div className={clipboardClass}>
+      {hideTextarea === true ? (
+        <textarea defaultValue={text} style={{ display: 'none' }} />
+      ) : (
+        <textarea
+          defaultValue={text}
+          onChange={({ target: { value } }) => setText(value)}
+          {...textareaProps}
+        />
+      )}
```

Comment:
If we have `hideTextarea` possibility, why we still add it? Shouldn't be something like:
```suggestion
    <div className={clipboardClass}>
      {hideTextarea === true ? (
        null
      ) : (
        <textarea
          defaultValue={text}
          onChange={({ target: { value } }) => setText(value)}
          {...textareaProps}
        />
      )}
```

Reply:
thank you.